llvm/include/llvm/ADT/FunctionExtras.h

```diff
@@ -317,8 +318,10 @@ template <typename ReturnT, typename... ParamTs> class UniqueFunctionBase {
     // Clear the old callback and inline flag to get back to as-if-null.
     RHS.CallbackAndInlineFlag = {};
 
-#ifndef NDEBUG
-    // In debug builds, we also scribble across the rest of the storage.
+#if !defined(NDEBUG) && !LLVM_ADDRESS_SANITIZER_BUILD
```

Comment:
&& !LLVM_MEMORY_SANITIZER_BUILD # because memset makes uninitialzied stuff initialzed 
&& !LLVM_HWADDRESS_SANITIZER_BUILD   # I hope we will do similar to asan std:: support this year

Reply:
Actually unrelated, lets not do in this patch